pkg/detector/detector.go

```diff
@@ -340,17 +358,15 @@ func (d *ResourceDetector) OnUpdate(oldObj, newObj interface{}) {
 		return
 	}
 
+	if !eventfilter.SpecificationChanged(unstructuredOldObj, unstructuredNewObj) {
+		klog.V(4).Infof("Ignore update event of object (kind=%s, %s/%s) as specification no change", unstructuredOldObj.GetKind(), unstructuredOldObj.GetNamespace(), unstructuredOldObj.GetName())
+		return
+	}
```

Comment:
Is this any specials reasons for moving these lines forward?

Reply:
Most of the time, it exits from this if statement, reducing the number of type assertions after the if statement.

Reply:
got it